sdlib/d/gc/util.d

```diff
@@ -288,3 +292,46 @@ unittest alignUp {
 		}
 	}
 }
+
+/**
+ * A 'packedU15' represents a 15-bit unsigned integer residing in one or two bytes.
+ * It is always written in 'big-endian' orientation, with the least-significant
+ * byte (always present) occupying position 'loc + 1', and the most-significant
+ * byte (may or may not be present) occupying position 'loc'.
+ * The least-significant bit of the upper byte indicates the presence of the
+ * lower byte; its remaining 7 bits store the least-significant 7 bits of the
+ * value. The lower byte, if present, stores the most significant 8 bits.
+ */
+
+ushort readPackedU15(void* loc) {
+	// FIXME: detect that we're actually on a little-endian machine:
+	auto data = swapEndian(*(cast(ushort*) loc));
```

Comment:
There is no such thing and endianess on integers, but in how they are represented in memory. What you want is a `loadBigEndian` and `storeBigEndian` and/or `byteSwap`. On that note, there is already an intrinsic to byte swap.